app/src/main/kotlin/com/wire/android/util/permission/CallingRecordAudioRequestFlow.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.android.util.permission
+
+import android.content.Context
+import androidx.activity.compose.ManagedActivityResultLauncher
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import com.wire.android.util.extension.checkPermission
+
+@Composable
+fun rememberCallingRecordAudioRequestFlow(
+    onAudioPermissionGranted: () -> Unit,
+    onPermissionDenied: () -> Unit,
+): CallingAudioRequestFlow {
+    val context = LocalContext.current
+
+    val requestPermissionLauncher: ManagedActivityResultLauncher<Array<String>, Map<String, @JvmSuppressWildcards Boolean>> =
```

Comment:
I think a better name fo this would be requestBluetoothAndRecordAUdioPermission, that way we do not have to jump into CallingAudioRequestFlow class to see what permission are we requesting